src/api/ledger/transactions.js

```diff
@@ -1,102 +1,74 @@
 /* eslint-disable valid-jsdoc */
 'use strict';
 const _ = require('lodash');
-const assert = require('assert');
 const async = require('async');
 const utils = require('./utils');
+const parseTransaction = require('./parse/transaction');
 const validate = utils.common.validate;
 const errors = utils.common.errors;
 
 const DEFAULT_RESULTS_PER_PAGE = 10;
+const MIN_LEDGER_VERSION = 32570;
 
-/**
- * Retrieve a transaction from the Remote based on the account and hash
- *
- * Note that if any errors are encountered while executing this function
- * they will be sent back to the client through the res. If the query is
- * successful it will be passed to the callback function
- *
- * @global
- * @param {Remote} remote
- *
- * @param {RippleAddress} account
- * @param {Hex-encoded String|ASCII printable character String} identifier
- * @param {Object} options
- * @param {Function} callback
- *
- * @callback
- * @param {Error} error
- * @param {Transaction} transaction
- */
-function getTransaction(api, account, identifier, requestOptions, callback) {
-  try {
-    assert.strictEqual(typeof requestOptions, 'object');
-    validate.address(account);
-    validate.identifier(identifier);
-    validate.options(requestOptions);
-  } catch(err) {
-    return callback(err);
-  }
-
-  const options = {};
-  options.hash = identifier;
-
-  const isLedgerRangeRequest = !_.isUndefined(requestOptions.min_ledger)
-    && !_.isUndefined(requestOptions.max_ledger);
-
-  if (isLedgerRangeRequest) {
-    const minLedger = Number(options.min_ledger);
-    const maxLedger = Number(options.max_ledger);
-    for (let i = minLedger; i <= maxLedger; i++) {
-      if (!api.remote.getServer().hasLedger(i)) {
-        return callback(new errors.NotFoundError('Ledger not found'));
-      }
+function hasCompleteLedgerRange(remote, options) {
+  const minLedger = Number(options.minLedger || MIN_LEDGER_VERSION);
+  const maxLedger = Number(options.maxLedger || remote.getLedgerSequence());
+  for (let i = minLedger; i <= maxLedger; i++) {
+    if (!remote.getServer().hasLedger(i)) { // TODO: optimize this
```

Comment:
Yeah, I think we need one where ranges are merged on add, then a method that takes a range and checks if it is fully contained in one of the existing ranges.